dcm4che-imageio/src/main/java/org/dcm4che3/imageio/plugins/dcm/ParseEntireFileMetaData.java

```diff
@@ -0,0 +1,191 @@
+//
+///////////////////////////////////////////////////////////////
+//                C O P Y R I G H T  (c) 2019                //
+//        Agfa HealthCare N.V. and/or its affiliates         //
+//                    All Rights Reserved                    //
+///////////////////////////////////////////////////////////////
+//                                                           //
+//       THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF      //
+//        Agfa HealthCare N.V. and/or its affiliates.        //
+//      The copyright notice above does not evidence any     //
+//     actual or intended publication of such source code.   //
+//                                                           //
+///////////////////////////////////////////////////////////////
+//
+package org.dcm4che3.imageio.plugins.dcm;
+
+import org.dcm4che3.data.*;
+import org.dcm4che3.error.NoPixelDataException;
+import org.dcm4che3.image.PhotometricInterpretation;
+import org.dcm4che3.imageio.stream.BulkURIImageInputStream;
+import org.dcm4che3.imageio.stream.ByteArrayImageInputStream;
+import org.dcm4che3.io.stream.ImageInputStreamLoader;
+import org.dcm4che3.io.stream.ServiceImageInputStreamLoader;
+import org.dcm4che3.io.stream.DelegateImageInputStreamLoader;
+import org.dcm4che3.imageio.stream.ImageInputStreamAdapter;
+import org.dcm4che3.imageio.stream.SegmentedImageInputStream;
+import org.dcm4che3.io.BulkDataDescriptor;
+import org.dcm4che3.io.DicomInputStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.imageio.stream.ImageInputStream;
+import javax.imageio.stream.MemoryCacheImageInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.ByteOrder;
+import java.nio.file.Path;
+
+/**
+ * Load the entire file into memory as BulkData.  Pixels are streamed from the BulkData URI to clients.  This class
+ * sacrifices initial parse time for having a stateless class which is able to access the same file from different threads
+ * without synchronization.
+ *
+ */
```

Comment:
The initial implementation just reads the whole file.  I'm planning on extracting the code that was necessary for reading incrementally through the pixel data in a new DicomMetaData implementation in a subsequent release.  My goal is to enable implementations to customize the metadata handling to efficiently handle some of the very large instances that are being generated by modalities today.  

Reading the entire file and reading to the pixel data are two obvious implementations, but i'm hoping that vendors find new ways to override the metadata reading in the future.  It's not even necessary to start with Part 10 files:  we could conceivably write a Metadata implementation that is based on retrieval of JSON from webservices...